modules/registry/registry.go

```diff
@@ -203,3 +253,67 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 
 	return c, nil
 }
+
+// SetDockerAuthConfig sets the DOCKER_AUTH_CONFIG environment variable with
+// authentication for with the given host, username and password.
+// It returns the base64 encoded credentials and a function to reset back
+// to the previous state.
+func SetDockerAuthConfig(host, username, password string, additional ...string) (creds string, cleanup func(), err error) { //nolint:nonamedreturns // Adds context to the return values.
+	creds, auth, err := DockerAuthConfig(host, username, password, additional...)
+	if err != nil {
+		return "", nil, fmt.Errorf("docker auth config: %w", err)
+	}
+
+	previousAuthConfig := os.Getenv("DOCKER_AUTH_CONFIG")
+	os.Setenv("DOCKER_AUTH_CONFIG", auth)
+
+	cleanup = func() {
+		if previousAuthConfig == "" {
+			os.Unsetenv("DOCKER_AUTH_CONFIG")
+			return
+		}
+		os.Setenv("DOCKER_AUTH_CONFIG", previousAuthConfig)
+	}
+
+	return creds, cleanup, nil
+}
+
+// DockerAuthConfig returns the base64 encoded credentials and the auth config
+// for the provided details. It also accepts additional host, username and password
+// triples to add more auth configurations.
+func DockerAuthConfig(host, username, password string, additional ...string) (creds, authConfig string, err error) { //nolint:nonamedreturns // Adds context to the return values.
```

Comment:
Same here about the need of nonamed returns:
```suggestion
func DockerAuthConfig(host, username, password string, additional ...string) (creds, authConfig string, err error) { // Adds context to the return values.
```

Reply:
You're right I'm used to a more strict lint set.

We should discuss changes to the linters and config going forward, as there's a lot of code smells which could be caught by the linter, however would need a clean up pass first.

Reply:
Indeed, happy to discuss about that 😊